src/file_lines.rs

```diff
@@ -0,0 +1,221 @@
+//! This module contains types and functions to support formatting specific line ranges.
+use std::{cmp, iter, str};
+
+use itertools::Itertools;
+use multimap::MultiMap;
+use rustc_serialize::{self, json};
+
+use codemap::LineRange;
+
+/// A range that is inclusive of both ends.
```

Comment:
Oh yeah, I guess you should be clear about these being 0 or 1 indexed as well as the exclusivity.

I don't think humans are likely to use this interface (rather they will use some tool as an intermediary), so I don't think it makes a difference either way in that sense. I don't feel strongly that it should be exclusive - whatever makes the code cleaner, really.